### PR TITLE
Stm32h7

### PR DESCRIPTION
**Pull request by Joe-Joe-Joe-Joe**
- Debug levels were changed to not conflict with STM32 definitions
- Probably want to merge Jason's PR first
- I made a number of edits to message_types.h. I took the liberty of modifying the payload board IDs as well, since I know what a pain that will be down the line. Not sure what the status of gencanlib is, perhaps these will be irrelevant. 
- Code is still technically untested. Side note, I eventually want to write a self-test using the internal loopback mode
- As I write this, I realize I could bind all of the HAL functions through function pointers, and that would solve the FIFO issue

- - -
This change is [Reviewable](https://reviewable.io/reviews/waterloo-rocketry/canlib/52)

### CODE REVIEW

File: can_common.c [hunk 1]
```diff
@@ -121,6 +122,24 @@ bool build_actuator_stat_msg(uint32_t timestamp,
...
+								 can_msg_t *output)
+{
+	if(!output) { return false; }
+
+	output->sid = MSG_ACT_ANALOG_CMD | BOARD_UNIQUE_ID;
+	write_timestamp_3bytes(timestamp, output);
+
+	memcpy(output->data+4, &actuator_cmd, sizeof(actuator_cmd));
+
+	output->data_len = 7;
```
Comment (Joe-Joe-Joe-Joe) [inline can_common.c:138]: Shouldn't this be length 8? 3 timestamp + 1 ID + 4 command value?

Comment (Joe-Joe-Joe-Joe) [reply can_common.c:138]: It also just doesn't populate the actuator ID field

File: can_common.c [hunk 2]
```diff
@@ -226,6 +245,24 @@ bool build_imu_data_msg(uint16_t message_type,
...
+bool build_state_est_data_msg(uint16_t message_type,
+                        uint32_t timestamp,
+                        float *data,   // floating point
+						enum STATE_ID data_id,
+                        can_msg_t *output)
+{
+	output->sid = MSG_STATE_EST_DATA | BOARD_UNIQUE_ID;
+	write_timestamp_3bytes(timestamp, output);
+	uint8_t *bytes = (uint8_t *)data;
+	output->data[3] = bytes[3];
```
Comment (Joe-Joe-Joe-Joe) [inline can_common.c:257]: We should come up with a consistent method (perhaps that's `memcpy`) for putting floats into messages, and then update the message definitions to match.

File: can_common.c [hunk 3]
```diff
@@ -388,6 +425,25 @@ bool build_gps_info_msg(uint32_t timestamp,
...
+						   uint16_t apogee,
+						   can_msg_t *output)
+{
+	if(!output) { return false; }
+
+	output->sid = MSG_STATE_EST_CALIB | BOARD_UNIQUE_ID;
+	write_timestamp_3bytes(timestamp, output);
+
+	output->data[3] = ack_flag;
+	output->data[4] = apogee & 0xff;
```
Comment (Joe-Joe-Joe-Joe) [inline can_common.c:439]: likewise, this byte ordering conflicts with whats in `message_types.h`

File: can_common.c [hunk 4]
```diff
@@ -492,6 +549,23 @@ int get_req_actuator_state(const can_msg_t *msg)
...
+float get_req_actuator_state_analog(const can_msg_t *msg)
+{
+    if (!msg) { return -1; }
+
+    uint16_t msg_type = get_message_type(msg);
+    switch (msg_type) {
+        case MSG_ACT_ANALOG_CMD:
+        	float value;
+        	memcpy(&value, msg->data+4, sizeof(value));
+            return msg->data[5];
```
Comment (Joe-Joe-Joe-Joe) [inline can_common.c:561]: should we not return `value`???

File: can_common.c [hunk 5]
```diff
@@ -606,6 +682,22 @@ bool get_imu_data(const can_msg_t *msg, uint16_t *output_x, uint16_t *output_y,
...
 }
 
+bool get_state_est_data(const can_msg_t *msg, float *data, enum STATE_ID *data_id){
+	if(!msg) { return false; }
+	if(!data) { return false; }
+	if(!data_id) { return false; }
+	if(get_message_type(msg) != MSG_STATE_EST_DATA) { return false; }
+
+	uint8_t *bytes = (uint8_t*)data;
+	bytes[3] = msg->data[3];
```
Comment (Joe-Joe-Joe-Joe) [inline can_common.c:692]: okay I think we should use `memcpy`

**Review by JasonBrave — APPROVED** [review]
LGTM

**Pull request merged**